src/coreclr/tools/Common/JitInterface/CorInfoTypes.cs

```diff
@@ -1509,12 +1509,52 @@ public struct CORINFO_SWIFT_LOWERING
         private struct SwiftLoweredTypes
```

Comment:
Should this be only comparing `numLoweredElements` worth of items?

It would require the Equals and GetHashCode to be on the `CORINFO_SWIFT_LOWERING` that I think makes more sense in general.